src/test/unit/common/languageserver/languageserver.test.ts

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import {
+  ExtensionContext,
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+} from '../../../../snyk/common/vscode/types';
+import SecretStorageAdapter from '../../../../snyk/common/vscode/secretStorage';
+import sinon from 'sinon';
+import { IConfiguration, PreviewFeatures } from '../../../../snyk/common/configuration/configuration';
+import assert, { deepStrictEqual } from 'assert';
+import { LanguageServer } from '../../../../snyk/common/languageserver/languageserver';
+import { ILanguageClientAdapter } from '../../../../snyk/common/vscode/languageClient';
+import { IVSCodeWorkspace } from '../../../../snyk/common/vscode/workspace';
+import { stubWorkspaceConfiguration } from '../../mocks/workspace.mock';
+
+suite('languageserver', () => {
+  let extensionContext: ExtensionContext;
+  let configuration: IConfiguration;
+  let languageServer: LanguageServer;
+  setup(() => {
+    extensionContext = {
```

Comment:
We can create a mock for it in `mocks` folder and share it between this and `configuration.test.ts`. It may be used somewhere else afterwards. See if it makes sense, no strict opinion here, just trying to simplify test setup.

Reply:
I'm not even sure we need it...